### ISSUE TITLE
patch fails to install

### ISSUE THREAD

**Issue by naushit** [skycocker/chromebrew#952]
Aarch = aarch64

hronos@localhost ~ $ crew install patch
patch: Patch takes a patch file containing a difference listing produced by the diff program and applies those differences to one or more original files, producing patched versions.
http://savannah.gnu.org/projects/patch/
version 2.7.5
No precompiled binary available for your platform, downloading source...
--2017-07-23 11:56:16--  https://ftp.gnu.org/gnu/patch/patch-2.7.5.tar.xz
Resolving ftp.gnu.org... 208.118.235.20
Connecting to ftp.gnu.org|208.118.235.20|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 727704 (711K) [application/x-tar]
Saving to: 'patch-2.7.5.tar.xz'

patch-2.7.5.tar.xz               100%[=========================================================>] 710.65K   398KB/s    in 1.8s    

2017-07-23 11:56:18 (398 KB/s) - 'patch-2.7.5.tar.xz' saved [727704/727704]

Archive downloaded
Unpacking archive, this may take a while...
gcc (GCC) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking whether make supports nested variables... (cached) no
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... none
checking dependency style of gcc... none
checking for gcc option to accept ISO C99... -std=gnu99
checking for gcc -std=gnu99 option to accept ISO Standard C... (cached) -std=gnu99
checking how to run the C preprocessor... /lib/cpp
configure: error: in `/usr/local/tmp/crew/patch-2.7.5.tar.xz.dir/patch-2.7.5':
configure: error: C preprocessor "/lib/cpp" fails sanity check
See `config.log' for more details
patch failed to install: `./configure --prefix=/usr/local` exited with 1

**Comment by uberhacker** [comment]
@naushit: Can you please share the contents of config.log?
```
preprocessor "/lib/cpp" fails sanity check Seeconfig.log' for more details
```

**Comment by jam7** [comment]
It looks like similar to #654.  Didn't you install `gcc` directly instead of `buildessential`?  I guess you need to install `buildessential`.

**Comment by naushit** [comment]
@jam7 

You are right, installing buildessential  solves problem.  Is it documented somewhere?  it should be part of  install script.  

Just my 2 cents.

**Comment by jam7** [comment]
if you type `crew install patch` just after installing chromebrew, chromebrew guides you to install buildessential.  we doesn't cover the scenario like installing gcc by hand unfortunately.

but, we can write this in faq. :-)

**Comment by jam7** [comment]
Added information to https://github.com/skycocker/chromebrew/wiki/FAQ.

**Comment by naushit** [comment]
This is great and thanks for the update

But I think most people will follow README.md from github page ( install section ) .  We should add 2 liners  there.  If we can  this part of install.sh , it's even better. 

Again, Thanks you all.

**Comment by jam7** [comment]
If you didn't install `gcc` by hand, chromebrew guides you like below.  Isn't it good enough?
```
$ wget -q -O - https://raw.github.com/skycocker/chromebrew/master/install.sh | bash
...
$ crew install patch
...
You don't have a working C compiler. Run 'crew install buildessential' to get one and try again.
```
If you still need to do something, you can make a PR to modify patch.rb to add `depends_on 'buildessential' => :build` line to requires `buildessentiall` at compile time.  Almost all packages should have that line, but many of us just forget it at the creation of new package files .

**Comment by uberhacker** [comment]
@jam7: I have to agree with @naushit here.  It would be better if all the buildessential dependencies were installed initially when chromebrew is installed.  I think I understand why you want it to be separate.  Eventually, we will offer binaries of all the packages and not need to compile, correct?  However, currently we need to compile most of the packages and it would be better if those dependencies were installed initially instead of requiring a `depends_on 'buildessential'` line in every package.

**Comment by jam7** [comment]
@uberhacker if it didn't takes several ten minutes to install buildessential, I could agree.

**Comment by uberhacker** [comment]
It might take 10 minutes to install but that is far less than the wasted time of wrestling with errors without any clue what is the cause.  Besides, that is the first thing I'm prompted to do after I install crew anyway.  +1 to add these dependencies in the install.sh script.

**Comment by jam7** [comment]
There are similar problems in #972 recently.  Maybe we should install `buildessential` in install.sh.  Please submit PR, @uberhacker.

**Comment by cstrouse** [comment]
Fixed by #1217.
